plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1411 @@
+                    for have_enable_traps in delete_trap:
+                        if have_enable_traps == 'all' and 'all' in have_enable_trap:
+                            enable_trap_url = "data/ietf-snmp:snmp/ietf-snmp-ext:system/trap-enable"
+                            enable_trap_request = {"path": enable_trap_url, "method": DELETE}
+                            enable_trap_requests.append(enable_trap_request)
+                        if have_enable_traps == 'link-down' and 'link-down' in have_enable_trap:
+                            enable_trap_url = "data/ietf-snmp:snmp/ietf-snmp-ext:system/notifications/link-down-trap"
+                            enable_trap_request = {"path": enable_trap_url, "method": DELETE}
+                            enable_trap_requests.append(enable_trap_request)
+                        if have_enable_traps == 'link-up' and 'link-up' in have_enable_trap:
+                            enable_trap_url = "data/ietf-snmp:snmp/ietf-snmp-ext:system/notifications/link-up-trap"
+                            enable_trap_request = {"path": enable_trap_url, "method": DELETE}
+                            enable_trap_requests.append(enable_trap_request)
+                        if have_enable_traps == 'config-change' and 'config-change' in have_enable_trap:
+                            enable_trap_url = "data/ietf-snmp:snmp/ietf-snmp-ext:system/notifications/config-change-trap"
+                            enable_trap_request = {"path": enable_trap_url, "method": DELETE}
+                            enable_trap_requests.append(enable_trap_request)
+                        if have_enable_traps == 'ospf' and 'ospf' in have_enable_trap:
+                            enable_trap_url = "data/ietf-snmp:snmp/ietf-snmp-ext:system/notifications/ospf-traps"
+                            enable_trap_request = {"path": enable_trap_url, "method": DELETE}
+                            enable_trap_requests.append(enable_trap_request)
+                        if have_enable_traps == 'bgp' and 'bgp' in have_enable_trap:
+                            enable_trap_url = "data/ietf-snmp:snmp/ietf-snmp-ext:system/notifications/bgp-traps"
+                            enable_trap_request = {"path": enable_trap_url, "method": DELETE}
+                            enable_trap_requests.append(enable_trap_request)
+                        if have_enable_traps == 'auth-fail' and 'auth-fail' in have_enable_trap:
+                            enable_trap_url = "data/ietf-snmp:snmp/ietf-snmp-ext:system/notifications/authentication-failure-trap"
+                            enable_trap_request = {"path": enable_trap_url, "method": DELETE}
+                            enable_trap_requests.append(enable_trap_request)
+                if enable_trap_requests:
+                    enable_trap_requests_list.extend(enable_trap_requests)
+
+        if delete_all or engine:
+            if delete_all or have_engine and configs['engine'] == have_engine:
```

Comment:
The config/diff filtering of the input deletion requests should have left only matching deletion requests in "configs". So if "engine" is in "configs", it should have the same value as "have_engine".

Is there a reason why this extra check is needed?

```suggestion
            if have_engine is not None:
```